packages/blade/src/components/DatePicker/_decisions/decisions.md

```diff
@@ -0,0 +1,352 @@
+
+## Anatomy
+
+<img src="./datepicker-anatomy.png" width="100%" alt="DatePicker Anatomy" />
+
+## Components
+
+- Calendar
+- DatePicker
+
+## Basic Usage
+
+```jsx
+<DatePicker
+  label="Pick date"
+  selectionType="single"
+  valueFormat="DD/MM/YYYY"
+  defaultValue={new Date()}
+  onChange={(date) => {
+    console.log('date selected', date);
+  }}
+/>
+```
+
+## API
+
+### DatePicker API
+
+```ts
+type InputProps = Pick<
+  BaseInputProps,
+  | 'label'
+  | 'labelPosition'
+  | 'name'
+  | 'validationState'
+  | 'errorText'
+  | 'successText'
+  | 'helpText'
+  | 'necessityIndicator'
+  | 'isRequired'
+  | 'isDisabled'
+  | 'onFocus'
+  | 'onBlur'
+  | 'size'
+  | 'autoFocus'
+>;
+
+type DatePickerProps = InputProps &
+  CalendarProps & {
+    /**
+     * Sets the format of the date value
+     *
+     * single selection @default 'DD/MM/YYYY'
+     * range selection @default 'DD/MM/YYYY - DD/MM/YYYY'
+     */
+    valueFormat: string;
+  };
+```
+
+### CalendarProps
+
+```ts
+type PickerType = 'date' | 'month' | 'year';
+type DateRange = [Date, Date];
+type Preset = {
+  /**
+   * Label for the preset
+   *
+   * @example 'Last 7 days'
+   */
+  label: string;
+  /**
+   * Value can be a single date or a range of dates
+   *
+   * @example with single selection
+   * dayjs().add(7, 'day')
+   *
+   * @example with range selection
+   * [dayjs().subtract(7, 'day'), dayjs()]
+   */
+  value: Date | DateRange;
+};
+
+type CalendarProps = {
+  /**
+   * Sets the selection mode of the calendar
+   * @default 'single'
+   */
+  selectionType: 'single' | 'range';
+  /**
+   * Sets the picker type
+   * @default 'date'
+   */
+  picker: PickerType;
+  onPickerChange: (picker: PickerType) => void;
+
+  // Standard controlled/uncontrolled state props
+  value: Date | DateRange;
+  defaultValue: Date | DateRange;
+  onChange: (value: Date | DateRange) => void;
+
+  /**
+   * Defines presets for the date picker
+   *
+   * @example with single selection
+   * [
+   *   { label: 'In 7 days', value: dayjs().add(7, 'day') },
+   *   { label: 'In a month', value: dayjs().add(1, 'month') },
+   * ]
+   *
+   * @example with range selection
+   * [
+   *   { label: 'Last 7 days', value: [dayjs().subtract(7, 'day'), dayjs()] },
+   *   { label: 'Last month', value: [dayjs().subtract(1, 'month'), dayjs()] },
+   * ]
+   */
+  presets: Preset[];
+  /**
+   * Sets the first day of the week in the calendar.
+   *
+   * 0-6 (0=Sunday, 1=Monday, etc.)
+   * @default 1
+   */
+  firstDayOfWeek: number;
+  /**
+   * Sets the minimum date that can be selected.
+   */
+  minDate?: Date;
+  /**
+   * Sets the maximum date that can be selected.
+   */
+  maxDate?: Date;
+  /**
+   * Determines whether single date can be selected as range, applicable only when type="range"
+   * @default false
+   */
+  allowSingleDateInRange?: boolean;
+  /**
+   * Sets the locale for the calendar.
+   *
+   * @default 'en'
+   */
+  locale: string;
+
+  // Basic selection props
```

Comment:
How will these callbacks work? Won't some of them overlap? For example when you change the month it will trigger - onMonthChange, onNext, onNextMonth (in edge cases it may also trigger onYearChange, onNextYear?)

Can we not just keep onChange and let users compute what has been changed?

Reply:
We are not going to provide onMonthChange/onYearChange. 

Only: 

  onNext: (date: Date) => void;
  onNextMonth: (date: Date) => void;
  onNextYear: (date: Date) => void;
  onNextDecade: (date: Date) => void;
  onPrevious: (date: Date) => void;
  onPreviousYear: (date: Date) => void;
  onPreviousMonth: (date: Date) => void;
  onPreviousDecade: (date: Date) => void;
  
  
  > Can we not just keep onChange and let users compute what has been changed?
  
  That will not be a very DX friendly way to provide event handling information. 
  
  Plus our OnChange only triggers when you click on "Apply" button it doesn't trigger if you click the Chevrons.

Reply:
Mostly analytics and controlled state management.

Reply:
Update: we decided to remove the redundant onNextYear,onNextDecade,onNextMonth etc props in favour of 1 onNext prop. 

```ts
onNext?: (args: { date: Date; type: "month" | "year" | "decade" }) => void;
```